spirv_glsl.cpp

```diff
@@ -1037,6 +1037,17 @@ void CompilerGLSL::emit_buffer_block(const SPIRVariable &var)
 	statement("");
 }
 
+void CompilerGLSL::emit_flattened_buffer_block(const SPIRVariable &var)
+{
+	auto &type = get<SPIRType>(var.basetype);
+
+	// Block names should never alias.
+	auto buffer_name = to_name(type.self, false);
+	size_t buffer_size = (get_declared_struct_size(type) + 15) / 16;
+
+	statement("uniform vec4 ", buffer_name, "[", buffer_size, "];");
```

Comment:
We could also check if the buffer block is read only, flattening SSBOs is just not going to happen :)